### PR TITLE
added simple update check

### PR DESCRIPTION
**Pull request by chrmoritz**
Do we want to do something like this?

### CODE REVIEW

File: starcheat/gui/mainwindow.py [hunk 1]
```diff
@@ -57,11 +58,27 @@ def __init__(self):
 
         logging.info("Main window init")
 
+        # check for updates
+        try:
+            latest_tag = request.urlopen("https://github.com/wizzomafizzo/starcheat/releases/latest").geturl()
```
Comment (chrmoritz) [inline starcheat/gui/mainwindow.py]: `https://github.com/wizzomafizzo/starcheat/releases/latest` redirects always to the latest published tag

**Comment by chrmoritz** [comment]
If could offer a option in the error handler to upload the log to gist too. @wizzomafizzo What do you think?

**Comment by wizzomafizzo** [comment]
i think both would be really useful, definitely a good idea. my only thing to add is it should handle offline use/connection errors nicely (probably silently fail) and perhaps configurable on/off in the options

i'm good to pull this immediately though (as long as we remember to flesh it out before next release)

**Comment by chrmoritz** [comment]
It already handles offline connections by catching the `URLError`. But I think we have some other design issues to discuss: for example where we should put this code.

File: starcheat/gui/mainwindow.py [hunk 2]
```diff
@@ -57,11 +59,27 @@ def __init__(self):
...
+                dialog = QMessageBox()
+                dialog.setWindowTitle("Outdated starcheat version")
+                dialog.setText("A new version of starcheat is available! Do you want to update now?")
+                dialog.setStandardButtons(QMessageBox.Yes | QMessageBox.No)
+                dialog.setDefaultButton(QMessageBox.Yes)
+                dialog.setIcon(QMessageBox.Question)
+                if dialog.exec() == QMessageBox.Yes:
+                    webbrowser.open(latest_tag, 2)
+                    sys.exit(0)
+        except URLError:
```
Comment (chrmoritz) [inline starcheat/gui/mainwindow.py]: thrown if user is offline

**Comment by wizzomafizzo** [comment]
i would say just put it in utils.py as a separate function at the top, make it accept the dialog object as a param (and use that as the parent for QMessageBox)

**Comment by chrmoritz** [comment]
So I actual wasted the last hour finding out, that there is no way in python to make a POST request with JSON data in python except building the HTTP package by yourself or installing a third party module like request (I hate you urllib).
But at least the update check should be fine now.

**Pull request closed without merging**